### PR TITLE
Issue 148: add newtype for column and row indices.

### PR DESCRIPTION
**Pull request by LukeTemp**
This PR adds a newtype for column and row to improve the type safety and readability of the code: https://github.com/qrilka/xlsx/issues/148

### CODE REVIEW

File: src/Codec/Xlsx/Types/Common.hs [hunk 1]
```diff
@@ -42,8 +41,8 @@ module Codec.Xlsx.Types.Common
   , DateBase(..)
   , dateFromNumber
   , dateToNumber
-  , int2col
-  , col2int
+  , columnIndexToText -- previously int2col
```
Comment (qrilka) [inline src/Codec/Xlsx/Types/Common.hs:44]: what about marking previous ones as deprecated and making them aliases to the new ones before dropping in a later release?

**Review by qrilka — COMMENTED** [review]
The change looks good to me and mostly a mechanical one but would you mind fixing CI failure?

**Comment by qrilka** [comment]
Also it looks like the merged in #158 brought some conflicts with this one

File: src/Codec/Xlsx/Types/Common.hs (hunk 1 above)
Comment (qrilka) [reply src/Codec/Xlsx/Types/Common.hs:46]: @LukeTemp do you still plan to work on this?

Comment (LukeTemp) [reply src/Codec/Xlsx/Types/Common.hs:46]: Yes @qrilka I plan to get this finished tomorrow, I was having issues with resolving 3 test cases that I did not expect to be affected by these changes - I'll elaborate in my commit

File: src/Codec/Xlsx/Types/Internal/CommentTable.hs [hunk 2]
```diff
@@ -75,6 +75,10 @@ parseComment authors cur = do
     authorId <- cur $| attribute "authorId" >=> decimal
     let author = fromJustNote "authorId" $ M.lookup authorId authors
     return (ref, Comment txt author True)
+    -- ^ Why is this always True? In testXlsx, we specify some comments as visible
+    -- (e.g., bob) and some as invisible (e.g., alice). I don't understand how these
+    -- test cases (particularly 'toXlsxEither: properly formatted') were passing
+    -- before the refactor.
```
Comment (LukeTemp) [inline src/Codec/Xlsx/Types/Internal/CommentTable.hs:81]: There are 3 test cases that I wasn't able to resolve: `toXlsxEither: properly formatted`, `write . read == id` and `write . fast-read == id`. They all fail based on some data in testXlsx, but I'm confused as to why the test case didn't fail before these changes. As part of the conversions carried out in these tests we expect to get `comment2 = Comment (XlsxRichText [rich1, rich2]) "alice" False` (line 194 in TestXlsx.hs). This comment should be invisible, but in the code above we are always defining comments as visible?

Comment (LukeTemp) [reply src/Codec/Xlsx/Types/Internal/CommentTable.hs:81]: Fixing this in another PR

Comment (LukeTemp) [reply src/Codec/Xlsx/Types/Internal/CommentTable.hs:81]: Other PR is here, I'd expect this to resolve the test case failures here: https://github.com/qrilka/xlsx/pull/163

File: src/Codec/Xlsx/Types/Common.hs [hunk 3]
```diff
@@ -89,9 +94,27 @@ import Data.Profunctor(dimap)
...
+  deriving (Eq, Ord, Show, Read, Generic, Num, Real, Enum, Integral)
+newtype ColumnIndex = ColumnIndex {unColumnIndex :: Int}
+  deriving (Eq, Ord, Show, Read, Generic, Num, Real, Enum, Integral)
+instance NFData RowIndex
+instance NFData ColumnIndex
+
+instance ToAttrVal RowIndex where
+  toAttrVal = toAttrVal . unRowIndex
+
+ -- | DEPRECATED: this function will be removed in an upcoming release, use
```
Comment (qrilka) [inline src/Codec/Xlsx/Types/Common.hs:107]: What about using pragmas instead? https://downloads.haskell.org/~ghc/latest/docs/users_guide/exts/pragmas.html?#warning-and-deprecated-pragmas - I think they are automatically added to haddocks as well

File: test/StreamTests.hs [hunk 4]
```diff
@@ -141,8 +144,11 @@ simpleWorkbookRow :: Xlsx
 simpleWorkbookRow = def & atSheet "Sheet1" ?~ sheet
   where
     sheet = toWs [((1,1), a1), ((2,1), cellValue ?~ CellText "text at A2 Sheet1" $ def)]
+--    sheets = [("Sheet1" , toWs
+--      [ ((RowIndex 1, ColumnIndex 1), a1)
+--      , ((RowIndex 2, ColumnIndex 1), cellValue ?~ CellText "text at A2 Sheet1" $ def)])]
```
Comment (qrilka) [inline test/StreamTests.hs:149]: why do we have these lines and also a similar  commeted out `sheets`  below?

Comment (LukeTemp) [reply test/StreamTests.hs:149]: Thanks for spotting - I must have left this in after rebasing

**Comment by qrilka** [comment]
thanks a lot

**Pull request merged**